extend config file inputs tests

bernt-matthias opened the pull request:
Shows that optional selects get `"None"` in the json file. I guess this needs to be changed to `None` -- i.e. its a bug. 

Any hint where the code is that generates this?

For files the result is analogous to what we deal with in cheetah. But I would `[]` still more intuitive than `[None]` for optional multiple selects. May be we can change this? But I'm fine with the state of the art .. since its at least consistent.

bernt-matthias committed:
extend config file inputs tests

add cases for ooptional/multiple data/select inputs

bernt-matthias commented:
Sorry .. I overwrote some recent changes because it mis-`git`ed things again :( will change this tomorrow.

bernt-matthias committed:
json configfiles: handle slect like data_column

- in particular the case of an empty select did yield "None" before
  now its []
  Problem was that the ValueWrapper is not None if nothing is selected
- but also non empty selects were "option1,option2" now its
  ["option1","option2"]
- renamed variable value -> value_wrapper

jmchilton commented on lib/galaxy/tools/parameters/wrapped_json.py:
```diff
@@ -79,20 +79,25 @@ def _data_input_to_path(v):
+        json_value = _cast_if_not_none(value_wrapper, bool)
+    elif input_type == "select":
+        if input.multiple:
+            json_value = [_ for _ in _cast_if_not_none(value_wrapper.value, list)]
```
If you can get the tests working, this change definitely makes sense to me!

bernt-matthias replied:
Guess this is possible. 

I just would like to use the opportunity to remark that this PR (in addition to fix the "None" problem for selects) also changes the following:

- multiple selects are now lists (before comma separated string)

Also I'm not sure if data_column should be cast to int, because of `<param type="data_column" 
use_header_names="true".../>` (but I could not get this attribute working as expected anyway).

bernt-matthias committed:
column_param_configfile test: don't check json content

dont compare the json file content but rely completely on the assertions on
the loaded json contents.

the json content test was superficial and did not work anymore since the order
of the keys differed. it worked before since the inputs contained only a
single key.

bernt-matthias committed:
remove remnant cat

mvdbeek reviewed (approved):
Looks good to me, are you happy with this @bernt-matthias ?

bernt-matthias commented:
Yes.
